### PR TITLE
kondo 0.4 (new formula)

### PR DESCRIPTION
**Pull request by tbillington**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Not exactly sure how I should test this, it scans filepaths for software projects. Possibly generate a dummy project and look for it?

Also curious if there's a way to distribute this as a build? I've seen other Rust projects do this but I can't see anywhere in the docs how to achieve it.

### CODE REVIEW

File: Formula/kondo.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  desc "Save disk space by cleaning non-essential files from software projects"
+  homepage "https://github.com/tbillington/kondo"
+  url "https://github.com/tbillington/kondo/archive/v0.4.tar.gz"
+  sha256 "f5044d744e3eb0db815c521537a34cfbead18bd7d5df0f6f5312a8c4f72f682e"
+  license "MIT"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", "--locked", "--root", prefix, "--path", "kondo"
```
Comment (dawidd6) [inline Formula/kondo.rb:11]:
```suggestion
    cd "kondo"
      system "cargo", "install", *std_cargo_args
    end
```

**Comment by dawidd6** [comment]
> Not exactly sure how I should test this, it scans filepaths for software projects. Possibly generate a dummy project and look for it?

Please do.

> Also curious if there's a way to distribute this as a build? I've seen other Rust projects do this but I can't see anywhere in the docs how to achieve it.

I'm not sure I understand, please elaborate.

**Comment by tbillington** [comment]
Ah, turns out what I meant was bottles (https://docs.brew.sh/Bottles) and just found out that they're generated, I don't make them!

Ok, will look into generating a test project and scanning it 🙂

File: Formula/kondo.rb (hunk 1 above)
Comment (tbillington) [reply Formula/kondo.rb:11]: Ah yes, much nicer :)

**Comment by tbillington** [comment]
Ok I've updated the test to use kondo to delete a generated project! It's the first time I've ever touched Ruby, hopefully it's roughly correct :)

**Comment by dawidd6** [comment]
I force-pushed some changes. Also, there is no need for such excessive commenting, but overall I think the formula is fine now.

**Comment by tbillington** [comment]
Apologies, the comments are for me in case I need to come back in 3 months 😅

**Review by dawidd6 — APPROVED** [review]

**Pull request closed without merging**